blockchain/chain_sync/src/sync.rs

```diff
@@ -65,7 +66,7 @@ pub struct ChainSyncer<DB, TBeacon> {
     beacon: Arc<TBeacon>,
 
     /// manages retrieving and updates state objects
-    state_manager: StateManager<DB>,
+    state_manager: Arc<RwLock<StateManager<DB>>>,
```

Comment:
There exists a check_block_msgs which doesn't contain the self reference and that method takes in Tbeacon which cannot be shared between threads and if we want the cache to actually work while it is in the scope of the sync object. We could probably use this statemanger without the need for self if we don't have the hashmap.

Might be better just to remove the cache

Reply:
I wasn't suggesting removing the Arc, which I think you may be assuming, I'm just saying remove the lock on the whole state manager and just wrap the cache in it

Reply:
Sorry I don't think I explained it right.

I meant you have a tipset_state that takes in a mutable reference in order to update that cache. Through the validations though, that check_block_msgs takes in the statemanger which also updates again using tipset_state. I might be wrong but with arcs we can use get_mut but it panics whenever more than reference is holding to it. 

In this case we have a shared reference that needs a write opretion, the future that is going to use this statemanger to await and the code under the construction of this future that both call tipset_state.

Unless I am understanding you wrong that is

Reply:
I'm saying you don't need a mutable reference for tipset_state because it only updates the cache? If the cache is wrapped in a mutex you don't need mutable ref to update

Reply:
Oh I see what you are saying now, I thinking in an async vs sync lock perspective. I will switch it